src/dmd/expressionsem.d

```diff
@@ -6700,28 +6736,10 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                          * variable with a bit copy of the default
                          * initializer
                          */
-                        Expression einit;
-                        if (sd.zeroInit && !sd.isNested())
-                        {
-                            // https://issues.dlang.org/show_bug.cgi?id=14606
-                            // Always use BlitExp for the special expression: (struct = 0)
-                            einit = new IntegerExp(exp.loc, 0, Type.tint32);
-                        }
-                        else if (sd.isNested())
-                        {
-                            auto sle = new StructLiteralExp(exp.loc, sd, null, t1);
-                            if (!sd.fill(exp.loc, sle.elements, true))
-                                return setError();
-                            if (checkFrameAccess(exp.loc, sc, sd, sle.elements.dim))
-                                return setError();
+                        Expression einit = getInitExp(sd, exp.loc, sc, t1);
+                        if (einit.op == TOK.error)
+                            return setError();
```

Comment:
This is redundant.  `getInitExp` returned an `ErrorExp` and then `setError` immediately creates a new `ErrorExp` and assigns it to result.  I see few options:

- create an overload of `setError` to take an `ErrorExp`
- instead of `return setError` do `result = einit; return;`

Maybe there is another solution, but those are the two that come to mind for me.

Reply:
Done